Connectors/CSi/Speckle.Connectors.ETABSShared/HostApp/ETABSSendCollectionManager.cs

```diff
@@ -0,0 +1,117 @@
+using Speckle.Connectors.CSiShared.HostApp;
+using Speckle.Converters.Common;
+using Speckle.Converters.CSiShared;
+using Speckle.Sdk.Models;
+using Speckle.Sdk.Models.Collections;
+
+namespace Speckle.Connectors.ETABSShared.HostApp;
+
+/// <summary>
+/// ETABS-specific collection manager that organizes structural elements by story and type.
+/// </summary>
+/// <remarks>
+/// Creates a two-level hierarchy:
+/// 1. Story level (e.g., "Story 1", "Story 2")
+/// 2. Element type level with ETABS-specific categorization:
+///    - Frames: Columns, Beams, Braces
+///    - Shells: Walls, Floors, Ramps
+///
+/// Elements without story assignment are placed in "Unassigned" collection.
+/// Uses caching to maintain collection references and prevent duplicates.
+/// </remarks>
+public class ETABSSendCollectionManager : CSiSendCollectionManager
+{
+  public ETABSSendCollectionManager(IConverterSettingsStore<CSiConversionSettings> converterSettings)
+    : base(converterSettings) { }
+
+  // TODO: This is gross. Too many strings. Improve as part of next milestone. Out of scope of "First Send".
+  public override Collection AddObjectCollectionToRoot(Base convertedObject, Collection rootObject)
+  {
+    var properties = convertedObject["properties"] as Dictionary<string, object>;
+    string story = GetStoryName(properties);
+    string objectType = GetObjectType(convertedObject, properties);
+
+    var storyCollection = GetOrCreateStoryCollection(story, rootObject);
+
+    var typeCollection = GetOrCreateTypeCollection(objectType, storyCollection);
+
+    return typeCollection;
+  }
+
+  // TODO: This is gross. Too many strings. Improve as part of next milestone. Out of scope of "First Send".
+  private Collection GetOrCreateStoryCollection(string story, Collection rootObject)
+  {
+    string storyPath = $"Story_{story}";
+    if (CollectionCache.TryGetValue(storyPath, out Collection? existingCollection))
+    {
+      return existingCollection;
+    }
+
+    var storyCollection = new Collection(story);
+    rootObject.elements.Add(storyCollection);
+    CollectionCache[storyPath] = storyCollection;
+    return storyCollection;
+  }
+
+  // TODO: This is gross. Too many strings. Improve as part of next milestone. Out of scope of "First Send".
+  private Collection GetOrCreateTypeCollection(string objectType, Collection storyCollection)
+  {
+    string typePath = $"{storyCollection["name"]}_{objectType}";
+    if (CollectionCache.TryGetValue(typePath, out Collection? existingCollection))
+    {
+      return existingCollection;
+    }
+
+    var typeCollection = new Collection(objectType);
+    storyCollection.elements.Add(typeCollection);
+    CollectionCache[typePath] = typeCollection;
+    return typeCollection;
+  }
+
+  // TODO: This is gross. Too many strings. Improve as part of next milestone. Out of scope of "First Send".
+  private string GetObjectType(Base convertedObject, Dictionary<string, object>? properties)
+  {
```

Comment:
a bit confused on why we mutate properties on collection manager instead of converter

Reply:
Good point. Probably better to add a collection parameter during the conversions and just simply use that on the collection manager to group (and not mix design concerns)